keras_cv/callbacks/pycoco_callback.py

```diff
@@ -20,83 +19,88 @@
 
 
 class PyCOCOCallback(Callback):
-    def __init__(self, validation_data, bounding_box_format, **kwargs):
+    def __init__(self, validation_data, bounding_box_format, cache=True, **kwargs):
+        """Creates a callback to evaluate PyCOCO metrics on a validation dataset.
+
+        Args:
+            validation_data: a tf.data.Dataset containing validation data. Entries
+                should have the form ```(images, {"gt_boxes": boxes,
+                "gt_classes": classes})```.
+            bounding_box_format: the KerasCV bounding box format used in the
+                validation dataset (e.g. "xywh")
+            cache: whether the callback should cache the dataset between iterations.
+                Note that if the validation dataset has shuffling of any kind
+                (e.g from `shuffle_files=True` in a call to TFDS.load or a call
+                to tf.data.Dataset.shuffle() with `reshuffle_each_iteration=True`),
+                you **must** cache the dataset to preserve iteration order. This
+                will store your entire dataset in main memory, so for large datasets
+                consider avoiding shuffle operations and passing `cache=False`.
+        """
         self.model = None
         self.val_data = validation_data
+        if cache:
```

Comment:
why this cannot be cached when building the pipeline?
```
eval_ds = ...
eval_ds = eval_ds.cache()
```

Reply:
If cache is for performance related, instead of determinism, maybe this is fine? Not related to coco, but for other evaluation dataset, it's generally better to be cached

Reply:
Caching by default is to enforce determinism. The purpose of structuring the API this way is to give users the power to control caching and determinism themselves if they want.